Source/Csla.Blazor.WebAssembly.Tests/SessionManagerTests.cs

```diff
@@ -0,0 +1,204 @@
+             };
+           }
+         })
+         .Verifiable();
+
+      // use real http client with mocked handler here
+      var httpClient = new HttpClient(handlerMock.Object)
+      {
+        BaseAddress = new Uri("http://test.com/"),
+      };
+      return httpClient;
+    }
+
+    private static byte[] GetSession(SessionMessage session, ApplicationContext _applicationContext)
+    {
+      var info = new MobileFormatter(_applicationContext).SerializeToDTO(session);
+      var ms = new MemoryStream();
+      new CslaBinaryWriter(_applicationContext).Write(ms, info);
+      ms.Position = 0;
+      var array = ms.ToArray();
+      return array;
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithTimeoutValue_ShouldNotThrowException()
+    {
+      var timeout = TimeSpan.FromHours(1);
+      var session = await _sessionManager.RetrieveSession(timeout);
+      Assert.AreEqual(session, SessionValue.Session);
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithZeroTimeout_ShouldThrowTimeoutException()
+    {
+      var timeout = TimeSpan.Zero;
+      await Assert.ThrowsExceptionAsync<TimeoutException>(() => _sessionManager.RetrieveSession(timeout));
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithValidCancellationToken_ShouldNotThrowException()
+    {
+      var cts = new CancellationTokenSource();
+      var session = await _sessionManager.RetrieveSession(cts.Token);
+      Assert.AreEqual(session, SessionValue.Session);
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithCancelledCancellationToken_ShouldThrowTaskCanceledException()
+    {
+      var cts = new CancellationTokenSource();
+      cts.Cancel();
+      await Assert.ThrowsExceptionAsync<TaskCanceledException>(() => _sessionManager.RetrieveSession(cts.Token));
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithTimeoutValue_ShouldNotThrowException()
+    {
+      var timeout = TimeSpan.FromHours(1);
+      await _sessionManager.SendSession(timeout);
+      Assert.IsTrue(true);
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithZeroTimeout_ShouldThrowTimeoutException()
+    {
+      var timeout = TimeSpan.Zero;
+      await Assert.ThrowsExceptionAsync<TimeoutException>(() => _sessionManager.SendSession(timeout));
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithValidCancellationToken_ShouldNotThrowException()
+    {
+      var cts = new CancellationTokenSource();
+      await _sessionManager.SendSession(cts.Token);
+      Assert.IsTrue(true);
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithCancelledCancellationToken_ShouldThrowTaskCanceledException()
+    {
+      var cts = new CancellationTokenSource();
+      cts.Cancel();
+      await Assert.ThrowsExceptionAsync<TaskCanceledException>(() => _sessionManager.SendSession(cts.Token));
+    }
+
+
+    [TestMethod]
+    public void  RetrieveCAchedSessionSession()
```

Comment:
Shouldn't we have two tests here? One checks for `null` return when no session is cached and one which returns a session when it's expected a session is cached?
Without anything this test currently only tests that no exception is thrown in "fresh startup" situation, right?